tip-distributor/src/claim_mev_workflow.rs

```diff
@@ -32,121 +41,426 @@ pub enum ClaimMevError {
-            }
+            panic!("Expected to have at least {desired_balance} lamports in {payer_pubkey}. Current balance is {start_balance} lamports. Deposit {sol_to_deposit} SOL to continue.");
         }
-        let stake_acct_min_rent = rpc_client.get_minimum_balance_for_rent_exemption(StakeState::size_of()).await.expect("Failed to calculate min rent");
-        let mut below_min_rent_count: usize = 0;
-        let mut zero_lamports_count: usize = 0;
-        for tree in merkle_trees.generated_merkle_trees {
-            // only claim for ones that have merkle root on-chain
-            let account = rpc_client.get_account(&tree.tip_distribution_account).await.expect("expected to fetch tip distribution account");
-            let fetched_tip_distribution_account = TipDistributionAccount::try_deserialize(&mut account.data.as_slice()).expect("failed to deserialize tip_distribution_account state");
-            if fetched_tip_distribution_account.merkle_root.is_none() {
-                info!(
-                        "not claiming because merkle root isn't uploaded yet. skipped {} claimants for tda: {:?}",
-                        tree.tree_nodes.len(),
-                        tree.tip_distribution_account
-                    );
+        let transactions_len = transactions.len();
+
+        info!("Sending {} tip claim transactions. {zero_lamports_count} would transfer zero lamports, {below_min_rent_count} would be below minimum rent", transactions.len());
+        let send_start = Instant::now();
+        let (remaining_transactions, new_failed_transaction_count) =
+            sign_and_send_transactions_with_retries_multi_rpc(
+                &keypair,
+                &blockhash_rpc_client,
+                &rpc_clients,
+                transactions,
+                max_loop_duration,
+            )
+            .await;
+
+        datapoint_info!(
+            "claim_mev_workflow-send_transactions",
+            ("transaction_count", transactions_len, i64),
+            (
+                "successful_transaction_count",
+                transactions_len - remaining_transactions.len(),
+                i64
+            ),
+            (
+                "remaining_transaction_count",
+                remaining_transactions.len(),
+                i64
+            ),
+            (
+                "failed_transaction_count",
+                new_failed_transaction_count,
+                i64
+            ),
+            ("send_latency_us", send_start.elapsed().as_micros(), i64),
+        );
+
+        if remaining_transactions.is_empty() {
+            info!("Finished claiming tips. {max_loop_retries} retries. {} remaining mev claim transactions, {failed_transaction_count} failed requests.", remaining_transactions.len());
+            return Ok(());
+        }
+
+        failed_transaction_count += new_failed_transaction_count;
+        retries += 1;
+        if retries >= max_loop_retries {
+            panic!(
```

Comment:
in the spirit of less panics, wdyt we return Err(e)?

Reply:
updated